pkg/admission/customresourcevalidation/config/deny_delete_cluster_config_resource.go

```diff
@@ -0,0 +1,52 @@
+package config
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/apiserver/pkg/admission"
+
+	"github.com/openshift/api/config"
+)
+
+const PluginName = "config.openshift.io/DenyDeleteClusterConfigurationResource"
+
+// Register registers an admission plugin factory whose plugin prevents the deletion of cluster configuration resources.
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return newAdmissionPlugin(), nil
+	})
+}
+
+var _ admission.ValidationInterface = &admissionPlugin{}
+
+type admissionPlugin struct {
+	*admission.Handler
+}
+
+func newAdmissionPlugin() *admissionPlugin {
+	return &admissionPlugin{Handler: admission.NewHandler(admission.Delete)}
+}
+
+// Validate returns an error if there is an attempt to delete a cluster configuration resource.
+func (p *admissionPlugin) Validate(attributes admission.Attributes) error {
+	if len(attributes.GetSubresource()) > 0 {
+		return nil
+	}
+	if attributes.GetResource().Group != config.GroupName {
+		return nil
+	}
+	switch attributes.GetResource().Resource {
```

Comment:
@sanchezl comment on the general reasoning here

```
// clusteroperators can be deleted so that we can force status refreshes and change over time
// clusterversions other than `version` can be deleted because others shouldn't exist
// any other resource in config.openshift.io named cluster may not be deleted.

Reply:
Done